CalendarExternalContentProvider/app/src/main/java/shuvalov/nikita/calendarexternalcontentprovider/MainActivity.java

```diff
@@ -0,0 +1,74 @@
+package shuvalov.nikita.calendarexternalcontentprovider;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.provider.CalendarContract;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+public class MainActivity extends AppCompatActivity {
+    RecyclerView mRecyclerView;
+    CalendarRecyclerAdapter mAdapter;
+    ArrayList<Event> mEvents;
+    Cursor mCursor;
+    public static final int MY_PERMISSION_REQUEST = 404;
+    public static final int RIGHT_TO_WRITE_REQUEST = 7;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        mEvents = new ArrayList<>();
+
+        setupRecyclerView();
+
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.READ_CALENDAR) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this,new String[]{Manifest.permission.READ_CALENDAR}, MY_PERMISSION_REQUEST);
+        }else if (ActivityCompat.checkSelfPermission(this, Manifest.permission.READ_CALENDAR) == PackageManager.PERMISSION_GRANTED){
+            mCursor = getContentResolver().query(CalendarContract.Events.CONTENT_URI, null, null, null, CalendarContract.Events.DTSTART, null);
```

Comment:
You're using the content provider directly rather than using a `CursorLoader` - this works, but a `CursorLoader` has some advantages. Namely, once set up it continues monitoring the content provider for changes, so if any other app adds/deletes from the calendar, it will trigger an update and that data will be reflected in your app. You don't have that ability to automatically update with your approach.